<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="voxel_size" default="0.2"/>
  <arg name="voxels_per_side" default="16" />
  <arg name="truncation_distance" default="0.6" />
  <arg name="method" default="fast"/>
  <arg name="ray_length" default="5" />
  <arg name="min_ray_length" default="0.1" />

  <!-- defines name of the namespace of the drone -->
  <arg name="UAV_NAME" default="$(optenv UAV_NAME uav2)" />

  <arg name="world_frame" default="map" />

  <group ns="$(arg UAV_NAME)">

    <!--Depth camera PointCloud-->
    <node pkg="nodelet" type="nodelet" name="standalone_nodelet" args="manager" />

    <!--Re-register from depth camera to RGB camera-->
    <!--node pkg="nodelet" type="nodelet" name="register" args="load depth_image_proc/register standalone_nodelet">
      <remap from="rgb/camera_info" to="/$(arg UAV_NAME)/rgbd_front_pitched/color/camera_info"/>
      <remap from="depth/camera_info" to="/$(arg UAV_NAME)/rgbd_front_pitched/aligned_depth_to_color/camera_info"/>
      <remap from="depth/image_rect" to="/$(arg UAV_NAME)/rgbd_front_pitched/aligned_depth_to_color/image_raw"/>
      <remap from="depth_registered/camera_info" to="/$(arg UAV_NAME)/rgbd_front_pitched/aligned_depth_to_color/camera_info_tf"/>
      <remap from="depth_registered/image_rect" to="/$(arg UAV_NAME)/rgbd_front_pitched/aligned_depth_to_color/image_raw_tf"/>
    </node-->

    <!--Make Depth camera PointCloud Colour-->
    <!--node pkg="nodelet" type="nodelet" name="point_cloud_xyzrgb" args="load depth_image_proc/point_cloud_xyzrgb standalone_nodelet">
      <remap from="rgb/camera_info" to="/$(arg UAV_NAME)/rgbd_front_pitched/color/camera_info"/>
      <remap from="rgb/image_rect_color" to="/$(arg UAV_NAME)/rgbd_front_pitched/color/image_raw"/>
      <remap from="depth_registered/image_rect" to="/$(arg UAV_NAME)/rgbd_front_pitched/aligned_depth_to_color/image_raw_tf"/>
      <remap from="depth_registered/points" to="/$(arg UAV_NAME)/rgbd_front_pitched/aligned_depth_to_color/points"/>
    </node-->

    <!--Make Depth camera PointCloud-->
    <!--node pkg="nodelet" type="nodelet" name="point_cloud_xyz" args="load depth_image_proc/point_cloud_xyz standalone_nodelet">
      <remap from="camera_info" to="/$(arg UAV_NAME)/rgbd_front_pitched/aligned_depth_to_color/camera_info"/>
      <remap from="image_rect" to="/$(arg UAV_NAME)/rgbd_front_pitched/aligned_depth_to_color/image_raw"/>
      <remap from="points" to="/$(arg UAV_NAME)/rgbd_front_pitched/aligned_depth_to_color/points"/>
      <param name="queue_size" value="3" />
    </node-->

    <!--Voxblox Mapping-->
    <node name="voxblox_node" pkg="voxblox_ros" type="esdf_server" output="screen" args="-alsologtostderr">
      <!--remap from="pointcloud" to="/$(arg UAV_NAME)/pcl_filter_rs_front_pitched/points_processed"/-->
      <!--remap from="pointcloud" to="/$(arg UAV_NAME)/velodyne/scan"/-->
      <!--remap from="pointcloud" to="/$(arg UAV_NAME)/os_cloud_nodelet/points"/-->
      <!--remap from="voxblox_node/esdf_map_out" to="esdf_map" /-->
      <!--remap from="voxblox_node/tsdf_map_out" to="tsdf_map" /-->
      <param name="tsdf_voxel_size" value="$(arg voxel_size)" />
      <param name="tsdf_voxels_per_side" value="$(arg voxels_per_side)" />
      <!--param name="truncation_distance" value="$(arg truncation_distance)" /-->
      <param name="method" value="$(arg method)" />
      <param name="max_ray_length_m" value="$(arg ray_length)" />
      <!--param name="min_ray_length_m" value="$(arg min_ray_length)" /-->

      <param name="publish_tsdf_map" value="true" />
      <param name="publish_esdf_map" value="true" />

      <remap from="~esdf_map_in" to="voxblox_node/esdf_map_out" />
      <remap from="~tsdf_map_in" to="voxblox_node/tsdf_map_out" />

      <!--param name="mesh_filename" value="/home/joaomendes/gt_police_station.ply" /-->

      <!--param name="clear_sphere_for_planning" value="true" /-->
      <!--param name="clear_sphere_radius" value="1.2" /-->
      <!--param name="occupied_sphere_radius" value="10" /-->

      <param name="publish_pointclouds" value="true" />
      <param name="publish_tsdf_info" value="true" />
      <!--param name="color_ptcloud_by_weight" value="true" /-->

      <param name="use_freespace_pointcloud" value="false" />
      <!--remap from="freespace_pointcloud" to="/$(arg UAV_NAME)/freespace_pc_rs_front_pitched/points_freespace"/-->
      <!--param name="mesh_min_weight" value="1e-6" /-->
      <!--param name="color_mode" value="color" /-->
      <param name="color_mode" value="normals" />
      <param name="use_tf_transforms" value="true" />
      <param name="update_mesh_every_n_sec" value="1.0" />
      <param name="min_time_between_msgs_sec" value="0.0" />
      <!--param name="pointcloud_queue_size" value="1" /-->
      <param name="use_const_weight" value="false" />

      <!--param name="use_weight_dropoff" value="false" /-->
      <!--param name="use_sparsity_compensation_factor" value="true" /-->
      <!--param name="sparsity_compensation_factor" value="5.0f" /-->

      <param name="verbose" value="true" />
      <param name="world_frame" value="$(arg world_frame)" />
    </node>

  </group>

</launch>

